modules/c++/sys/source/OSUnix.cpp

```diff
@@ -346,6 +346,53 @@ void sys::OSUnix::getMemInfo(size_t &totalPhysMem, size_t &freePhysMem) const
 #endif
 }
 
+std::string sys::OSUnix::getCurrentExecutable(
+        const std::string& argvPathname) const
+{
+    std::vector<std::string> possibleSymlinks;
+
+    // Linux
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self", "exe")));
+
+    // Solaris
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self",
+            sys::Path::joinPaths("path", "a.out"))));
+
+    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)
+    {
+        struct stat linkStatus;
+        const std::string path = possibleSymlinks[ii];
+        if (lstat(path.c_str(), &linkStatus) == 0)
```

Comment:
If `lstat()` fails, we should throw.  Probably cleanest to add a function in the unnamed namespace that looks something like...

```
void linkStat(const std::string& pathname, stat& linkStatus);
```